### PR TITLE
Bug 1820474: clusterAutoscaler and machineAutoscaler structural schema

### PR DESCRIPTION
**Pull request by enxebre**
This makes clusterAutoscaler and machineAutoscaler structural schemas.
To this end I update controller-tools, which forced me to update the whole api machinery stack to kube 1.18. As a consequence I had to drop CVO to avoid transitive incompatible dependencies and take the chance to move to openshift/library-go. As a consequence I had to update scripts to run on golang 1.13.
Needs https://github.com/openshift/release/pull/8114.

### CODE REVIEW

**Comment by enxebre** [comment]
```
KIND:     ClusterAutoscaler
VERSION:  autoscaling.openshift.io/v1

DESCRIPTION:
     ClusterAutoscaler is the Schema for the clusterautoscalers API

FIELDS:
   apiVersion	<string>
   kind	<string>
   metadata	<Object>
      annotations	<map[string]string>
      clusterName	<string>
      creationTimestamp	<string>
      deletionGracePeriodSeconds	<integer>
      deletionTimestamp	<string>
      finalizers	<[]string>
      generateName	<string>
      generation	<integer>
      labels	<map[string]string>
      managedFields	<[]Object>
         apiVersion	<string>
         fieldsType	<string>
         fieldsV1	<map[string]>
         manager	<string>
         operation	<string>
         time	<string>
      name	<string>
      namespace	<string>
      ownerReferences	<[]Object>
         apiVersion	<string>
         blockOwnerDeletion	<boolean>
         controller	<boolean>
         kind	<string>
         name	<string>
         uid	<string>
      resourceVersion	<string>
      selfLink	<string>
      uid	<string>
   spec	<Object>
      balanceSimilarNodeGroups	<boolean>
      ignoreDaemonsetsUtilization	<boolean>
      maxPodGracePeriod	<integer>
      podPriorityThreshold	<integer>
      resourceLimits	<Object>
         cores	<Object>
            max	<integer>
            min	<integer>
         gpus	<[]Object>
            max	<integer>
            min	<integer>
            type	<string>
         maxNodesTotal	<integer>
         memory	<Object>
            max	<integer>
            min	<integer>
      scaleDown	<Object>
         delayAfterAdd	<string>
         delayAfterDelete	<string>
         delayAfterFailure	<string>
         enabled	<boolean>
         unneededTime	<string>
      skipNodesWithLocalStorage	<boolean>
   status	<map[string]>
```

```
KIND:     MachineAutoscaler
VERSION:  autoscaling.openshift.io/v1beta1

DESCRIPTION:
     MachineAutoscaler is the Schema for the machineautoscalers API

FIELDS:
   apiVersion	<string>
   kind	<string>
   metadata	<Object>
      annotations	<map[string]string>
      clusterName	<string>
      creationTimestamp	<string>
      deletionGracePeriodSeconds	<integer>
      deletionTimestamp	<string>
      finalizers	<[]string>
      generateName	<string>
      generation	<integer>
      labels	<map[string]string>
      managedFields	<[]Object>
         apiVersion	<string>
         fieldsType	<string>
         fieldsV1	<map[string]>
         manager	<string>
         operation	<string>
         time	<string>
      name	<string>
      namespace	<string>
      ownerReferences	<[]Object>
         apiVersion	<string>
         blockOwnerDeletion	<boolean>
         controller	<boolean>
         kind	<string>
         name	<string>
         uid	<string>
      resourceVersion	<string>
      selfLink	<string>
      uid	<string>
   spec	<Object>
      maxReplicas	<integer>
      minReplicas	<integer>
      scaleTargetRef	<Object>
         apiVersion	<string>
         kind	<string>
         name	<string>
   status	<Object>
      lastTargetRef	<Object>
         apiVersion	<string>
         kind	<string>
         name	<string>
```

**Comment by JoelSpeed** [comment]
L13 of the Makefile needs to go from `GOFLAGS += -mod=vendor` to `GOFLAGS ?= -mod=vendor` to work with Go 1.13, the flag is included by default from 1.13 onwards

File: pkg/controller/machineautoscaler/machineautoscaler_controller.go [hunk 1]
```diff
@@ -99,43 +100,64 @@ func (r *Reconciler) AddToManager(mgr manager.Manager) error {
...
-	for _, gvk := range missingGVKs {
-		r.RemoveSupportedGVK(gvk)
-	}
+	return nil
+}
 
-	// Fail if we didn't find any of the supported target types registered.
-	if len(r.config.SupportedTargetGVKs) < 1 {
-		return ErrNoSupportedTargets
+func getMissingGVKs(restMapper meta.RESTMapper, supportedCVKs []schema.GroupVersionKind) ([]schema.GroupVersionKind, error) {
```
Comment (elmiko) [inline pkg/controller/machineautoscaler/machineautoscaler_controller.go:136]: should the second argument be named `supportGVKs` to make it clearer?

File: pkg/controller/machineautoscaler/machineautoscaler_controller.go [hunk 2]
```diff
@@ -99,43 +100,64 @@ func (r *Reconciler) AddToManager(mgr manager.Manager) error {
...
+	var missingGVKs []schema.GroupVersionKind
+
+	for _, gvk := range supportedCVKs {
```
Comment (elmiko) [inline pkg/controller/machineautoscaler/machineautoscaler_controller.go:139]: `s/supportedCVKs/supportedGVKs/` ?

**Pull request merged**